soc/nordic/common/pinctrl_soc.h

```diff
@@ -25,21 +25,48 @@ extern "C" {
 /** Type for nRF pin. */
 typedef uint32_t pinctrl_soc_pin_t;
 
+/**
+ * @brief Utility macro to check if a function requires clock enable.
+ *
+ * @param node_id Node identifier.
+ * @param prop Property name.
+ * @param idx Property entry index.
+ * @param p_node_id Parent node identifier.
+ */
+#define Z_CHECK_CLOCK_ENABLE(node_id, prop, idx, fun) \
+	DT_PROP_BY_IDX(node_id, prop, idx) == fun ? BIT(NRF_CLOCK_ENABLE_POS) :
+
+/**
+ * @brief Utility macro compute the clock enable bit.
+ *
```

Comment:
maybe for future generations you can add here a comment why DT_FOREACH_PROP_ELEM_SEP_VARGS with empty separator must be used instead of DT_FOREACH_PROP_ELEM_VARGS as it might be confusing

Reply:
added comment